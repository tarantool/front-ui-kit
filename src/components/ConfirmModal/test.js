import React from 'react';
import renderer from 'react-test-renderer';
import { ConfirmModal } from './index';

jest.mock('react-dom', () => ({
  createPortal: node => node
}));

const action = jest.fn();

describe('ConfirmModal renders correctly', () => {
  it('hidden state', () => {
    const tree = renderer.create(
      <ConfirmModal
        title="Please confirm"
        visible={false}
        onCancel={action}
        onConfirm={action}
        confirmText="Remove"
      >
        Removing user John Appleseed
      </ConfirmModal>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('visible state', () => {
    const tree = renderer.create(
      <ConfirmModal
        title="Please confirm"
        visible={true}
        onCancel={action}
        onConfirm={action}
        confirmText="Remove"
      >
      Removing user John Appleseed
      </ConfirmModal>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('loading state', () => {
    const tree = renderer.create(
      <ConfirmModal
        title="Please confirm"
        visible={true}
        onCancel={action}
        onConfirm={action}
        confirmText="Remove"
        confirmPreloader={true}
      >
        Removing user John Appleseed
      </ConfirmModal>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});