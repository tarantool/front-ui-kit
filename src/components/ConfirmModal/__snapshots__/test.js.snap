// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal renders correctly hidden state 1`] = `null`;

exports[`ConfirmModal renders correctly loading state 1`] = `
<div
  className="css-18615gn"
  onMouseDown={[Function]}
>
  <div
    className="css-xynqqq"
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <h2
      className="css-34jfdi"
    >
      Please confirm
    </h2>
    <svg
      className="css-pjnldn"
      onClick={[MockFunction]}
      viewBox="0 0 16 16"
    >
      <use
        xlinkHref="#close"
      />
    </svg>
    <div
      className="css-i9gxme"
    >
      Removing user John Appleseed
    </div>
    <div
      className="css-vtqncq"
    >
      <div
        className="css-99wgdf"
      >
        <div
          className="css-12ibiio"
        >
          <button
            className="css-1oap1v"
            onClick={[MockFunction]}
            type="button"
          >
            Cancel
          </button>
        </div>
        <div
          className="css-12ibiio"
        >
          <button
            className="css-1dd4cey"
            onClick={[MockFunction]}
            type="button"
          >
            Remove
            <div
              className="css-9lvo7z"
            >
              <svg
                className="css-2ujzs2"
                viewBox="0 0 128 128"
              >
                <use
                  xlinkHref="#icon-spinner"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      className="css-vnc1z7"
      onFocus={[Function]}
      tabIndex="0"
    />
  </div>
</div>
`;

exports[`ConfirmModal renders correctly visible state 1`] = `
<div
  className="css-18615gn"
  onMouseDown={[Function]}
>
  <div
    className="css-xynqqq"
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <h2
      className="css-34jfdi"
    >
      Please confirm
    </h2>
    <svg
      className="css-pjnldn"
      onClick={[MockFunction]}
      viewBox="0 0 16 16"
    >
      <use
        xlinkHref="#close"
      />
    </svg>
    <div
      className="css-i9gxme"
    >
      Removing user John Appleseed
    </div>
    <div
      className="css-vtqncq"
    >
      <div
        className="css-99wgdf"
      >
        <div
          className="css-12ibiio"
        >
          <button
            className="css-1oap1v"
            onClick={[MockFunction]}
            type="button"
          >
            Cancel
          </button>
        </div>
        <div
          className="css-12ibiio"
        >
          <button
            className="css-8qjt25"
            onClick={[MockFunction]}
            type="button"
          >
            Remove
          </button>
        </div>
      </div>
    </div>
    <div
      className="css-vnc1z7"
      onFocus={[Function]}
      tabIndex="0"
    />
  </div>
</div>
`;
